### PR TITLE
Fix recursive header expansion error

### LINKED ISSUE #246: Recursive header expansion when using other libs such as Firebase
I installed AppCenter and CodePush as instructed. When I tried to compile I encountered this error:

```
Showing Recent Issues
Argument list too long: recursive header expansion failed at /Users/ethanwang/Github/reactnative/Glever_RNI/Glever/node_modules/appcenter/ios/../../../ios/Pods/React/node_modules/fsevents/node_modules/har-validator/lib/node6.
```
Enviroment: 

"react-native": "0.53.0",
node: v8.9.4
cocoapods-1.4.0

I have looked up this error it seems like it was fixed on V1.20 (https://github.com/Microsoft/AppCenter-SDK-React-Native/issues/209)

But I'm still experiencing the same error. How should I fix this?

### PR DESCRIPTION
**Pull request by bmourat**
Fixes #246

### CODE REVIEW

**Comment by dhei** [comment]
@bmourat VSTS build (https://msmobilecenter.visualstudio.com/_permalink/_build/index?collectionId=27bf89de-9697-418a-accf-1aa125b22914&projectId=c5683969-a866-4281-b5c4-3a18f2414bc6&buildId=153026) failed probably because there's a release in flight and the npm version not yet exist. Just wait a bit and trigger the build again after the release.

**Comment by bmourat** [comment]
@dhei thanks!

File: appcenter-analytics/ios/AppCenterReactNativeAnalytics.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -232,10 +232,16 @@
...
+					"$(SRCROOT)/../../../ios/Pods/AppCenter/**",
+					"$(SRCROOT)/../../../ios/Pods/AppCenterReactNativeShared/**",
 					"$(inherited)",
 				);
-				HEADER_SEARCH_PATHS = "$(SRCROOT)/../../react-native/React/**";
+				HEADER_SEARCH_PATHS = (
+					"$(SRCROOT)/../../react-native/React/Base",
+					"$(SRCROOT)/../../react-native/React/Modules",
+					"$(SRCROOT)/../../react-native/React/CSSLayout",
+					"$(SRCROOT)/../../react-native/React/Views",
```
Comment (jpstrikesback) [inline appcenter-analytics/ios/AppCenterReactNativeAnalytics.xcodeproj/project.pbxproj]: Adding "$(SRCROOT)/../../../ios/Pods/Headers/Public/React/**" or non recursive variants would solve issues where React is added via CocoaPods

**Comment by dhei** [comment]
Once #295 is merged, the Eslint error should go away.

File: appcenter-analytics/ios/AppCenterReactNativeAnalytics.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -232,10 +232,16 @@
...
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(SRCROOT)/../../../ios/Vendor/**",
-					"$(SRCROOT)/../../../ios/Pods/**",
+					"$(SRCROOT)/../../../ios/Pods/AppCenter/**",
+					"$(SRCROOT)/../../../ios/Pods/AppCenterReactNativeShared/**",
 					"$(inherited)",
 				);
-				HEADER_SEARCH_PATHS = "$(SRCROOT)/../../react-native/React/**";
+				HEADER_SEARCH_PATHS = (
+					"$(SRCROOT)/../../react-native/React/Base",
```
Comment (dhei) [inline appcenter-analytics/ios/AppCenterReactNativeAnalytics.xcodeproj/project.pbxproj]: Why narrowing down the header search paths from ** to 4 specific modules?

Comment (guperrot) [reply appcenter-analytics/ios/AppCenterReactNativeAnalytics.xcodeproj/project.pbxproj]: We can't have recursive at root as it causes https://github.com/Microsoft/AppCenter-SDK-React-Native/issues/246. So we try to limit to what we need.

**Pull request merged**